scaffoldmaker/utils/tubemesh2.py

```diff
@@ -0,0 +1,356 @@
+'''
```

Comment:
Merge this new implementation into tubemesh.py, replacing old one.
The original centrallinetube1 scaffold can be migrated or discarded. I'm guessing what it brought to the table was ellipse profiles with equal element spacing, so not sure how/whether to support?
What you have is likely able to get into the code for now, so following are largely future discussions:
Soon we will need variable wall thickness around and along units/segments, plus in each layer, and also controllable side axis or axes to produce twists. Here I'm thinking your approach of having unit side axes and having all sizing information in the unit/segments could work well, then e.g. ellipsoidal sections can be passed in.
Furthermore, eventually the nodes should be numbered from one end to the other, and elements too. However, xi1 can remain around, xi2 along and xi3 through the wall.
Same function should also support generating 2D meshes in future.